cmd/kubelet/app/options/options.go

```diff
@@ -337,7 +337,7 @@ func AddKubeletConfigFlags(fs *pflag.FlagSet, c *kubeletconfig.KubeletConfigurat
 	fs.DurationVar(&c.FileCheckFrequency.Duration, "file-check-frequency", c.FileCheckFrequency.Duration, "Duration between checking config files for new data")
 	fs.DurationVar(&c.HTTPCheckFrequency.Duration, "http-check-frequency", c.HTTPCheckFrequency.Duration, "Duration between checking http for new data")
 	fs.StringVar(&c.ManifestURL, "manifest-url", c.ManifestURL, "URL for accessing the container manifest")
-	fs.StringVar(&c.ManifestURLHeader, "manifest-url-header", c.ManifestURLHeader, "HTTP header to use when accessing the manifest URL, with the key separated from the value with a ':', as in 'key:value'")
+	fs.Var(flag.ColonSeparatedMultimapStringString(c.ManifestURLHeader), "manifest-url-header", "Comma-separated list of HTTP headers to use when accessing the manifest URL. Multiple headers with the same name will be added in the same order provided. For example: `a:hello,b:again,c:world,b:beautiful`")
```

Comment:
How about making this `StringArrayVar` so that we could use the flag multiple times, but setting only one header at a time? E.g., 
```
curl -H "Accept: application/xml" -H "Content-Type: application/xml" ...
```

Reply:
Hmm, I like that idea. I'll take a look at doing it.

Reply:
Offline chat with @yujuhong: using the flag multiple times is nice, but the downside is that you have to use []string instead of map[string][]string, which is less convenient from JSON/YAML perspective. We eventually plan to get rid of flags, so we'll stick with the map[string][]string solution.

Reply:
> using the flag multiple times is nice, but the downside is that you have to use []string instead of map[string][]string, which is less convenient from JSON/YAML perspective

I don't think that's true, I think it just means your `Set` impl has to handle being called multiple times. It is still free to populate any data structure it wants to

Reply:
@liggitt That's a good counterpoint to my argument, we could just stick with `fs.Var` and do whatever we want. 

Making Set iterative would also mean that flags append to defaults, rather than override them. In this specific case I suppose that would be ok, because we don't set any default headers. I can modify the Set implementation from this PR to stop it from clearing the previous values on each Set call. OTOH, we'll ultimately get rid of the flag, so maybe it doesn't matter that much. WDYT?

Reply:
> Making Set iterative would also mean that flags append to defaults, rather than override them

Only if you want it to (which we probably wouldn't). First call to Set would wipe existing value and toggle a flag that Set had been called.